test/plugin/test_out_lm.rb

```diff
@@ -147,5 +147,97 @@ def create_valid_subject
 
       assert_equal expected, result
     end
+
+    test "include_metadata true, try to include everything except in metadata_to_exclude" do
+      plugin = create_driver(%[
+        resource_mapping {"someProp": "lm_property"} 
+        include_metadata true
+        metadata_to_exclude ["meta1", "meta3","someProp"]
+      ]).instance
+      tag = "lm.test"
+      time = Time.parse("2020-08-23T00:53:15+00:00").to_i
+      record = {"message" => "Hello from test",  "timestamp" => "2020-10-30T00:29:08.629701504Z" , "meta1" => "testMeta1" , "meta2" => "testMeta2", "meta3" => "testMeta3", "meta4" => "testMeta4", "meta5" => {"key1" => "value1", "key2" => {"key2_1" => "value2_1"}}, "someProp" => "resourcePropVal" }
+
+      result = plugin.process_record(tag, time, record)
+        
+      expected = {
+          "message" => "Hello from test",
+          "timestamp" => "2020-10-30T00:29:08.629701504Z",
+          "_lm.resourceId" => {"lm_property" => "resourcePropVal"},
+          "meta2" => "testMeta2",
+          "meta4" => "testMeta4", 
+          "meta5" => {"key1" => "value1", "key2" => {"key2_1" => "value2_1"}},
+      }
+      assert_equal expected, result
+    end
+  end
+  
+
+  sub_test_case "device_less_logs" do
+    test "if device_less_logs is true, resource_mapping to be skipped and metadata to be added" do
+      plugin = create_driver(%[
+        resource_mapping {"a.b": "lm_property"} 
+        device_less_logs true
+      ]).instance
+      tag = "lm.test"
+      time = Time.parse("2020-08-23T00:53:15+00:00").to_i
+      record = {"message" => "Hello from test",  "timestamp" => "2020-10-30T00:29:08.629701504Z", "tag" => "lm.test", "meta1" => "metadata1", "meta2" => "metadata2", "service" => "lm-service", "namespace" => "lm-namepsace"}
+
+      result = plugin.process_record(tag, time, record)
+        
+      expected = {
+          "message" => "Hello from test",
+          "timestamp" => "2020-10-30T00:29:08.629701504Z",
+          "tag" => "lm.test",
+          "meta1" => "metadata1",
+          "meta2" => "metadata2",
+          "service" => "lm-service", 
+          "namespace" => "lm-namepsace"
+      }
+      assert_equal expected, result
+    end
+
+    test "if device_less_logs is true even when include_metadata false, skip metadata defined  in metadata_to_exclude and resource_mapping" do
+      plugin = create_driver(%[
+        resource_mapping {"a.b": "lm_property"} 
+        device_less_logs true
+        include_metadata false
+        metadata_to_exclude ["meta1", "meta3"]
```

Comment:
if metadata is disabled, then why are still sending metadata?
Then metadata_to_exclude should be irrelevant.